openedx/core/djangoapps/oauth_dispatch/jwt.py

```diff
@@ -170,7 +171,8 @@ def _create_jwt(
     # Default scopes should only contain non-privileged data.
     # Do not be misled by the fact that `email` and `profile` are default scopes. They
     # were included for legacy compatibility, even though they contain privileged data.
-    scopes = scopes or ['email', 'profile']
+    # The scope `user_id` must be added for requests with grant_type password.
+    scopes = _update_user_id_in_scopes(scopes or ['email', 'profile'], grant_type)
```

Comment:
Actually, looking more closely add the code, I think we should follow the same pattern of either using the explicitly defined scopes, or using the default scopes. Something like the following:
```
    # Default scopes should only contain non-privileged data.
    # Do not be misled by the fact that `email` and `profile` are default scopes. They
    # were included for legacy compatibility, even though they contain privileged data.
    if grant_type == Application.GRANT_PASSWORD:
        default_scopes = ['user_id', 'email', 'profile']
    else
        default_scopes = ['email', 'profile']
    scopes = scopes or default_scopes
```

All of this could be packaged into a private method if you wish, but it is more consistent (i.e. won't override explicitly set scopes) and keeps all the scopes code together. Thoughts?

Reply:
1. [Nit] Thanks for the clarification. This is just an opinion, but I'd prefer seeing `_update_user_id_in_scopes` renamed to `_get_updated_scopes` (or some name like that), and packaging all the scope related code (default + user_id) in one place. If you feel strongly otherwise, no need to update.
2. I can't think of a clean and simply way to create a new appropriate setting related to user_id, but wondering if we could at least add a comment before https://github.com/openedx/edx-platform/blob/7c25c5f4c857fcbd8da76f10f0e2aad5b7c53ab8/lms/envs/common.py#L1242 that reads something like:
```
# user_id is added in code as a default scope for JWT cookies and all password grant_type JWTs
```